EchoPro/survey.py

```diff
@@ -36,741 +25,457 @@ class Survey:
+        # any cracks, then this becomes an unchecked recursive function. Consequently, this
+        # should be leashed so that it can only dig for so long until it yields some sort of 
+        # Value or Runtime Error. 
+        return None     
+    
+    def load_survey_data( self ):
         """
-
-        if not params:
-            raise ValueError("You must provide parameters for the Kriging routine!")
-
-        # make sure all parameters are included
-        if set(krig_type_dict.keys()).difference(set(params.keys())):
-            raise ValueError("Some required parameters where not provided!")
-
-        # check that all types are correct for params
-        for key, val in params.items():
-            expected_type = krig_type_dict.get(key)
-            if not isinstance(val, expected_type):
-                raise TypeError(
-                    f"The Kriging parameter {key} is not of type {expected_type}"
-                )
-
-        krig = Kriging(
-            self,
-            params["k_max"],
-            params["k_min"],
-            params["R"],
-            params["ratio"],
-            params["s_v_params"],
-            params["s_v_model"],
-        )
-
-        return krig
-
-    def compute_kriging_results(self, kriging_params: dict) -> None:
+        Loads the biological, NASC, and stratification
+        data using parameters obtained from the configuration
+        files. This will generate data attributes associated with the tags
+        defined in both the configuration yml files and the reference CONFIG_MAP
+        and LAYER_NAME_MAP dictionaries.
+        """
+
+        ### Check whether data files defined from the configuration file exists
+        # Generate flat JSON table comprising all configuration parameter names
+        flat_configuration_table = pd.json_normalize(self.config)
+
+        # Parse the flattened configuration table to identify data file names and paths
+        parsed_filenames = flat_configuration_table.filter(regex="filename").values.flatten()
+
+        # Append the root directory to the parsed data file names
+        data_file_paths = [os.path.join(self.config['data_root_dir'], filename) for filename in parsed_filenames]
+
+        # Evaluate whether either file is missing
+        data_existence = [os.path.exists(file_path) for file_path in data_file_paths]
+
+        # Assign the existence status to each configuration file for error evaluation
+        data_status = dict(zip(data_file_paths, data_existence))
+
+        # Error evaluation and print message (if applicable)
+        if not all(data_status.values()):
+            missing_data = [data_file for data_file, exists in data_status.items() if not exists]
+            raise FileNotFoundError(f"The following data files do not exist: {missing_data}")
+        
+        ### Data validation and import
+        # Iterate through each file defined/detected in 'data_file_paths'
+        for file in data_file_paths:
```

Comment:
I read through what's in this entire for loop, and think that the code would be a lot cleaner if you do not try to write everything in a "one size fits all way" to check for layer depth and whether there's another layer above or below. Instead, I suggest hard coding which dict will be looking for `filename` and `sheetname` at the 3rd level (`biological`) and which dict will be looking for these at the 2nd level (`stratification`, `NASC`, `kriging`). This way the handling of that extra US/CAN level for the `biological` dict will also be explicit (and can be documented in the code comment and docstring). 

The two mapping dict `LAYER_NAME_MAP` and `CONFIG_MAP` both have simple structures and are completely under our control, and I think it is overkill to have the dedicated search/push/pull functions to deal with the 2nd or 3rd level indexing. These functions are doing something libraries that deals with XML trees are trying to achieve, but the "tree" structure we have here is small, I suggest to hard-code things when needed.

Along with factoring out 2 functions below ([here](https://github.com/uw-echospace/EchoPro/pull/161/files#r1443609137) and [here](https://github.com/uw-echospace/EchoPro/pull/161/files#r1443609380)), with a small amount of hard-coded selection rules, I think the code will be much easier to follow and debug/maintain later, since it directly makes use of the hierarchical structure in the yaml file.